hooks/useScrollController.ts

```diff
@@ -0,0 +1,36 @@
+import { useState, useEffect } from 'react';
+
+const useScrollPosition = () => {
+  const [isAtTop, setIsAtTop] = useState(false);
+  const [isAtMiddle, setIsAtMiddle] = useState(false);
+  const [isAtBottom, setIsAtBottom] = useState(false);
+
+  useEffect(() => {
+    const handleScroll = () => {
+      const scrollPosition = window.scrollY;
+      const windowHeight = window.innerHeight;
+      const documentHeight = document.documentElement.scrollHeight;
+
+      // 최상단 감지
+      const atTop = scrollPosition === 0;
+      setIsAtTop(atTop);
+
+      // 하단 감지
+      const atBottom = scrollPosition + windowHeight >= documentHeight - 10;
+      setIsAtBottom(atBottom);
+
+      // 중간 감지: 최상단도 하단도 아닌 경우
+      setIsAtMiddle(!atTop && !atBottom);
+    };
+
+    window.addEventListener('scroll', handleScroll);
+
+    return () => {
+      window.removeEventListener('scroll', handleScroll);
+    };
+  }, []);
+
+  return { isAtTop, isAtMiddle, isAtBottom };
+};
+
+export default useScrollPosition;
```

Comment:
요렇게 하나의 state로 관리하는 방식은 어떨까용?!

```suggestion
import { useState, useEffect } from 'react';

const useScrollPosition = () => {
  const [position, setPosition] = useState<'top' | 'middle' | 'bottom'>('top');

  useEffect(() => {
    const handleScroll = () => {
      const scrollPosition = window.scrollY;
      const windowHeight = window.innerHeight;
      const documentHeight = document.documentElement.scrollHeight;

      // 최상단 감지
      const atTop = scrollPosition === 0;

      if (atTop) {
        return setPosition('top');
      }

      // 하단 감지
      const atBottom = scrollPosition + windowHeight >= documentHeight - 10;

      if (atBottom) {
        return setPosition('bottom');
      }

      // 중간 감지: 최상단도 하단도 아닌 경우
      return setPosition('middle');
    };

    window.addEventListener('scroll', handleScroll);

    return () => {
      window.removeEventListener('scroll', handleScroll);
    };
  }, []);

  return { position };
};

export default useScrollPosition;

```

Reply:
[6b6dda9](https://github.com/dreamyPatisiel/devdevdev-client/pull/142/commits/6b6dda99c21eaff3e4629365f0941689f8385b18) 👍👍넘 좋네용